registry/storage/driver/swift/swift.go

```diff
@@ -0,0 +1,561 @@
+func (d *driver) Delete(ctx context.Context, path string) error {
+	opts := swift.ObjectsOpts{
+		Prefix: d.swiftPath(path),
+	}
+
+	objects, err := d.Conn.ObjectNamesAll(d.Container, &opts)
+	if err != nil {
+		return parseError(path, err)
+	}
+	if len(objects) == 0 {
+		return storagedriver.PathNotFoundError{Path: path}
+	}
+
+	if d.BulkDeleteSupport {
+		if _, err := d.Conn.BulkDelete(d.Container, objects); err != swift.Forbidden {
+			return parseError(path, err)
+		}
+	}
+
+	for _, name := range objects {
+		if _, headers, err := d.Conn.Object(d.Container, name); err == nil {
+			manifest, ok := headers["X-Object-Manifest"]
+			if ok {
+				components := strings.SplitN(manifest, "/", 2)
+				segContainer := components[0]
+				segments, err := d.getAllSegments(components[1])
+				if err != nil {
+					return parseError(name, err)
+				}
+
+				for _, s := range segments {
+					if err := d.Conn.ObjectDelete(segContainer, s.Name); err != nil {
+						return parseError(s.Name, err)
+					}
+				}
+			}
+		} else {
+			return parseError(name, err)
+		}
+
+		if err := d.Conn.ObjectDelete(d.Container, name); err != nil {
+			return parseError(name, err)
+		}
+	}
+
+	return nil
+}
+
+// URLFor returns a URL which may be used to retrieve the content stored at the given path.
+// May return an UnsupportedMethodErr in certain StorageDriver implementations.
+func (d *driver) URLFor(ctx context.Context, path string, options map[string]interface{}) (string, error) {
+	return "", storagedriver.ErrUnsupportedMethod
+}
+
+func (d *driver) swiftPath(path string) string {
+	return strings.TrimLeft(strings.TrimRight(d.Prefix+"/files"+path, "/"), "/")
+}
+
+func (d *driver) swiftSegmentPath(path string) string {
+	return strings.TrimLeft(strings.TrimRight(d.Prefix+"/segments"+path, "/"), "/")
+}
+
+func (d *driver) createParentFolders(path string) error {
+	dir := gopath.Dir(path)
+	for dir != "/" {
+		_, _, err := d.Conn.Object(d.Container, d.swiftPath(dir))
+		if err == swift.ContainerNotFound || err == swift.ObjectNotFound {
+			_, err := d.Conn.ObjectPut(d.Container, d.swiftPath(dir), bytes.NewReader(make([]byte, 0)),
+				false, "", directoryMimeType, nil)
+			if err != nil {
+				return parseError(dir, err)
+			}
+		}
+		dir = gopath.Dir(dir)
+	}
+
+	return nil
+}
+
+func (d *driver) getContentType() string {
+	return "application/octet-stream"
+}
+
+func (d *driver) getAllSegments(path string) ([]swift.Object, error) {
+	return d.Conn.Objects(d.Container, &swift.ObjectsOpts{Prefix: d.swiftSegmentPath(path)})
+}
+
+func (d *driver) createManifest(path string) (*swift.ObjectCreateFile, error) {
+	headers := make(swift.Headers)
+	headers["X-Object-Manifest"] = d.Container + "/" + d.swiftSegmentPath(path)
+	return d.Conn.ObjectCreate(d.Container, d.swiftPath(path), false, "",
+		d.getContentType(), headers)
+}
+
+func detectBulkDelete(authURL string) (bulkDelete bool) {
+	resp, err := http.Get(filepath.Join(authURL, "..", "..") + "/info")
```

Comment:
This should use path instead of filepath, otherwise it will use backslashes on windows